bin/init.js

```diff
@@ -30,56 +33,284 @@ const info = {
 		return `\x1b[34m${ message }\x1b[0m`;
 	},
 };
+
+/**
+ * Return root directory
+ *
+ * @return {string} root directory
+ */
+const getRoot = () => {
+	return path.resolve( __dirname, '../' );
+}
+
 let fileContentUpdated = false;
 let fileNameUpdated = false;
 let themeCleanup = false;
+let isGitInitialized = false;
 
 const args = process.argv.slice( 2 );
 
 if ( 0 === args.length ) {
 	rl.question( 'Would you like to setup the theme? (y/n) ', ( answer ) => {
+
 		if ( 'n' === answer.toLowerCase() ) {
 			console.log( info.warning( '\nTheme Setup Cancelled.\n' ) );
-			process.exit( 0 );
+			rl.close();
 		}
 
 		rl.question( 'Enter theme name (shown in WordPress admin)*: ', ( themeName ) => {
+
 			const themeInfo = renderThemeDetails( themeName );
+
 			rl.question( 'Confirm the Theme Details (y/n) ', ( confirm ) => {
+
 				if ( 'n' === confirm.toLowerCase() ) {
 					console.log( info.warning( '\nTheme Setup Cancelled.\n' ) );
-					process.exit( 0 );
+					rl.close();
 				}
 
 				initTheme( themeInfo );
 
-				rl.question( 'Would you like to run the theme cleanup? (y/n) ', ( cleanup ) => {
-					if ( 'n' === cleanup.toLowerCase() ) {
-						console.log( info.warning( '\nExiting without running theme cleanup.\n' ) );
-						process.exit( 0 );
+				// Remove theme initialization scripts.
+				modifyComposerJson();
+				modifyPackageJson();
+
+				rl.question( 'Would you like to initialize git (Note: It will delete any `.git` folder already in current directory)? (y/n) ', async ( initialize ) => {
+					if ( 'n' === initialize.toLowerCase() ) {
+						console.log( info.warning( '\nExiting without initializing GitHub.\n' ) );
+						await askQuestionForHuskyInstallation();
```

Comment:
That depends on the requirement. Let's say if they have initialized it manually and don't want it to initialize it again through the setup then this will allow to install husky. I have added conditions to properly display error message if git is not initialized.